### ISSUE TITLE
Error should be reported if @final is applied to non-method function

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6861]
The following should generate an error but does not.

```python
@final
def func1() -> int: # Type error: not allowed on non-method function.
    ...
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.345, which I just published. It will also be included in a future release of pylance.
